README.md

```diff
@@ -579,6 +580,314 @@ Note that the subflows are an array of repeating entries - one for each iteratio
+
+Your `pdf-map.yaml` file will consist of the following key value pairs:
+
+#### flow
+
+The name of the flow that this `pdf-map.yaml` file corresponds to.
+
+For example:
+
+```yaml
+flow: ubi
+```
+
+#### pdf
+
+The path to the pdf template file the FFB library should use when generating your PDF.
+The path should begin with a forward flash. For example `/pdfs/Example-PDF-File.pdf`.
+
+For example:
+
+```yaml 
+pdf: /pdfs/Multipage-UBI-Form.pdf
+```
+
+#### inputFields
+
+Key value pairs for your applications inputs and their corresponding PDF fields. Note that the key
+should always be the actual name of an input in your application and the value can be one of two
+things.
+
+For single value fields like text fields and radios, the value should be the name of the PDF field.
+For example:
+
+```yaml
+inputFields:
+  firstName: APPLICANT_LEGAL_NAME_FIRST
+  lastName: APPLICANT_LEGAL_NAME_LAST
+  genderIdentity: GENDER
+  birthDate: APPLICANT_DATE_OF_BIRTH
+```
+
+For multi-value fields like checkboxes, the value should be a map of key value pairs where the top
+level key is the name of the checkbox field and each internal key value pair is a map of checkbox
+field value to PDF field name. For example:
+
+```yaml
+inputFields:
+  programs:
+    SNAP: APPLICANT_PROGRAMS_SNAP
+    TANF: APPLICANT_PROGRAMS_TANF
+    CCAP: APPLICANT_PROGRAMS_CCAP
+  incomeTypes:
+    incomeJob: APPLICANT_HAS_JOB_INCOME
+    incomeSelf: APPLICANT_HAS_SELF_EMPLOYMENT_INCOME
+    incomeUnemployment: APPLICANT_HAS_UNEMPLOYMENT_INCOME
+    incomeSocialSecurity: APPLICANT_HAS_SOCIAL_SECURITY_INCOME
+```
+
+Note that for the multi-value checkbox field mappings, `programs` and `incomeTypes` represent the
+names
+of checkbox inputs in the application, and `SNAP`, `TANF`, `CCAP`, `incomeJob`, `incomeSelf`, etc
+are
+the actual values of checkboxes within those inputs. `APPLICANT_PROGRAMS_SNAP`
+and `APPLICANT_HAS_JOB_INCOME`
+are the names of the corresponding PDF checkbox fields.
+
+#### dbFields
+
+Key value pairs for database column fields from your application. The key is the name of a database
+column and the value is the name of a field in your PDF template. Currently supported fields are:
+
+- submittedAt
+- submissionId
+- createdAt
+- updatedAt
+- flow
+
+For example:
+
+```yaml
+dbFields:
+  submittedAt: SUBMISSION_TIME
+  submissionId: SUBMISSION_ID
+  flow: FLOW_NAME
+```
+
+### Subflow specific PDF field mappings
+
+#### subflowInfo
+
+The top level key for all your subflows. `subflowInfo` holds all data related to mapping all of your
+subflows.
+
+#### totalIterations
+
+The total number of iterations possible given your PDF's template. For example, if your PDF has room
+for 5 household members, your `totalIterations` for the `household` subflow would be 5.
+
+For example:
+
+```yaml
+subflowInfo:
+  household:
+    totalIterations: 5
+```
+
+#### fields
+
+Similar to `inputFields`, `fields` is a map of input fields from your application to PDF fields in
```

Comment:
So `fields` is the same thing as `inputFields`, just in the `subflowInfo` key? Could it be `inputFields`? Since it would be `subflowInfo.<subflowName>.inputFields`?

It's complicated either way, but my first thought was "what the heck are fields"?